packages/eas-cli/src/utils/profiles.ts

```diff
@@ -47,6 +57,45 @@ export async function getProfilesAsync<T extends ProfileType>({
   return await Promise.all(results);
 }
 
+function isBuildProfile(profile: any, type: ProfileType): profile is BuildProfile<Platform> {
+  return !!profile && type === 'build';
+}
+
+async function setNodeVersionFromFileAsync(
+  projectDir: string,
+  profile: BuildProfile<Platform>
+): Promise<void> {
+  const nodeVersion = await getNodeVersionFromFileAsync(projectDir);
+  if (nodeVersion) {
+    if (profile.node && profile.node !== nodeVersion) {
+      throw new Error(
+        `Node version defined in eas.json (${profile.node}) does not match the version defined in .nvmrc (${nodeVersion})`
+      );
+    }
+    Log.log(
+      `The EAS build profile does not specify a Node.js version. Using the version specified in .nvmrc: ${nodeVersion} `
+    );
+
+    profile.node = nodeVersion;
+  }
+}
+
+async function getNodeVersionFromFileAsync(projectDir: string): Promise<string | undefined> {
+  const nvmrcPath = path.join(projectDir, '.nvmrc');
+  try {
+    const nodeVersion = (await fs.readFile(nvmrcPath, 'utf8')).toString().trim();
+    if (!semver.valid(nodeVersion)) {
+      throw new Error(`Invalid node version in .nvmrc: ${nodeVersion}`);
+    }
+    return nodeVersion;
+  } catch (error: any) {
+    if (error.code !== 'ENOENT') {
+      Log.warn(`Failed to read node version from ${nvmrcPath}: ${error.message}`);
+    }
```

Comment:
I think it's better to check if a file exists before accessing it using `fs.exists` function